nvtabular/ops/groupby.py

```diff
@@ -199,18 +197,26 @@ def _compute_dtype(self, col_schema, input_schema):
             "sum": numpy.float32,
         }
 
-        is_lists = {"list": True}
+        agg = self._find_agg(col_schema, input_schema)
+        dtype = agg_dtypes.get(agg, col_schema.dtype)
+
+        return col_schema.with_dtype(dtype)
+
+    def _compute_shape(self, col_schema, input_schema):
+        agg_is_lists = {"list": True}
+
+        agg = self._find_agg(col_schema, input_schema)
+        is_list = agg_is_lists.get(agg, col_schema.is_list)
```

Comment:
In the case where we have fallback to the second argument of the `.get` here and check the col_schema.is_list attribute. If we have a fixed list, would we like to preserve that info in the shape later on instead of turning in into a ragged list which is presumably default shape corresponding to `DefaultShapes.LIST`

Reply:
I'm not actually sure. Are there `GroupBy` aggregations that don't change the shape of list columns?

Reply:
From looking at [the list of aggregations](https://github.com/NVIDIA-Merlin/NVTabular/blob/main/nvtabular/ops/groupby.py#L192-L202), I think everything changes the shape, either from list to scalar or from scalar to list.

Reply:
I wonder if the default there should actually be `False` 🤔

Reply:
In practice it seems that it may not matter whether it's `col_schema.is_list` or `False`. I tried running Groupby with a an agg that is not `"list"` on a dataframe with list feautres and we get errors in both cudf and pandas

```python
from merlin.io import Dataset
import nvtabular as nvt
import cudf

df = cudf.DataFrame({"a": [1, 1, 2], "b": [[10], [20], [20]]})
workflow = nvt.Workflow(["a", "b"] >> nvt.ops.Groupby(groupby_cols=["a"], aggs=["sum"]))
workflow.fit_transform(Dataset(df)).compute()
# => Raises DataError: All requested aggregations are unsupported.
``` 

Some of these aggs, like `sum` or `mean` could in theory work on list features if we wanted them to.

Pandas for example, handles sum across lists as concatenation.

```python
import pandas as pd

df = pd.DataFrame({"a": [1, 1, 2], "b": [[10], [20], [20]]})
df.groupby("a").sum()
# =>
          b
a          
1  [10, 20]
2      [20]

```

or if numpy arrays, then as an element-wise sum

```python
df = pd.DataFrame({"a": [1, 1, 2], "b": [np.array([10]), np.array([20]), np.array([20])]})
df.groupby("a").sum()
# =>
      b
a      
1  [30]
2  [20]
```

`var/std/mean/median` also work and in this example return scalars. If the element type contained an array of more then 1 dimension, then `mean` could start returning a list type too

since `cudf` doesn't appear to support aggregating across list columns this is probably something we don't need to be concerned about for now.

```python
import cudf
df = cudf.DataFrame({"a": [1, 1, 2], "b": [[10], [20], [20]]})
df.groupby("a").sum()
# => Raises DataError: All requested aggregations are unsupported.

df["b"].sum()
# => Raises TypeError: cannot perform sum with type list
```

Reply:
So I think `False` could be fine as the default here, and as far as this PR goes, it's no less clear than it was before.

If we need groupby aggregations for list columns as a future feature of NVTabular this will need to be revisited. I suppose even if cudf and pandas don't natively support this we could implement this ourselves by extracting the cupy/numpy arrays from the series in our own agg function to handle list column aggregations.

Reply:
That's great to know; I really appreciate your thoroughness in testing this out. This probably warrants a further update to the shapes here, I'll open a separate issue.

Reply:
Tracked here: https://github.com/NVIDIA-Merlin/NVTabular/issues/1763